app/components/specific/elevationProfile/ElevationProfile.js

```diff
@@ -100,11 +89,20 @@ export default class ElevationProfile extends Component {
   }
 
   static createAreaGraph(edges, graphWidth, graphHeight) {
-    const groupedEdgesByRange = edges.groupBy(groupEdgesByRange, {});
-
-    const groupedEdgesBySequence = {};
-    Object.entries(groupedEdgesByRange).forEach(([range, section]) => {
-      groupedEdgesBySequence[range] = section.groupBy(groupEdgesBySequence, []);
+    const groupedEdgesByRange = edges.groupBy(item => getRange(item.percent));
+
+    // eslint-disable-next-line
+    const data = Object.entries(groupedEdgesByRange).map(([grade, section]) => {
+      const indices = section.map(item => item.index);
+      const missingValueIndices = xor(
```

Comment:
those lines have the smell of dark magic

Reply:
Sure! Nothing to be proud of! Kind of #QuickWin!